lib/Server.js

```diff
@@ -625,6 +513,126 @@ class Server {
     });
   }
 
+  setupHttps() {
+    // if the user enables http2, we can safely enable https
+    if (this.options.http2 && !this.options.https) {
+      this.options.https = true;
+    }
+
+    if (this.options.https) {
+      // for keep supporting CLI parameters
+      if (typeof this.options.https === 'boolean') {
+        this.options.https = {
+          ca: this.options.ca,
+          pfx: this.options.pfx,
+          key: this.options.key,
+          cert: this.options.cert,
+          passphrase: this.options.pfxPassphrase,
+          requestCert: this.options.requestCert || false,
+        };
+      }
+
+      for (const property of ['ca', 'pfx', 'key', 'cert']) {
+        const value = this.options.https[property];
+        const isBuffer = value instanceof Buffer;
+
+        if (value && !isBuffer) {
+          let stats = null;
+
+          try {
+            stats = fs.lstatSync(fs.realpathSync(value)).isFile();
+          } catch (error) {
+            // ignore error
+          }
+
+          if (stats) {
+            // It is file
+            this.options.https[property] = fs.readFileSync(path.resolve(value));
+          } else {
+            this.options.https[property] = value;
+          }
+        }
+      }
+
+      let fakeCert;
+
+      if (!this.options.https.key || !this.options.https.cert) {
+        fakeCert = getCertificate(this.log);
+      }
+
+      this.options.https.key = this.options.https.key || fakeCert;
+      this.options.https.cert = this.options.https.cert || fakeCert;
+
+      // note that options.spdy never existed. The user was able
+      // to set options.https.spdy before, though it was not in the
+      // docs. Keep options.https.spdy if the user sets it for
+      // backwards compatibility, but log a deprecation warning.
+      if (this.options.https.spdy) {
+        // for backwards compatibility: if options.https.spdy was passed in before,
+        // it was not altered in any way
+        this.log.warn(
+          'Providing custom spdy server options is deprecated and will be removed in the next major version.'
+        );
+      } else {
+        // if the normal https server gets this option, it will not affect it.
+        this.options.https.spdy = {
+          protocols: ['h2', 'http/1.1'],
+        };
+      }
+    }
+
+    this.createServer();
```

Comment:
Let's move this after ` this.setupHttps();` (no in method)

Reply:
fixed